new-era-commands/slash/search.js

```diff
@@ -0,0 +1,67 @@
+const {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  SlashCommandBuilder,
+} = require("discord.js");
+
+const searchSites = {
+  google: {
+    name: "Google",
+    url: "https://www.google.com/search?q=",
+    description: "Search for terms in Google",
+  },
+  top: {
+    name: "TOP",
+    url: "https://www.google.com/search?q=site:theodinproject.com+",
+    description: "Search for terms in The Odin Project website with Google",
+  },
+  mdn: {
+    name: "MDN",
+    url: "https://developer.mozilla.org/en-US/search?q=",
+    description: "Search for terms in MDN",
+  },
+};
+
+const createCommand = (site) => (command) =>
+  command
+    .setName(site.name.toLowerCase())
+    .setDescription(site.description)
+    .addStringOption((option) =>
+      option
+        .setName("prompt")
+        .setDescription("term to search")
+        .setRequired(true),
+    )
+    .addUserOption((option) =>
+      option.setName("user").setDescription("user to ping"),
+    );
```

Comment:
```suggestion
const searchSlashCommand = new SlashCommandBuilder()
  .setName('search')
  .setDescription('Search in a specific site');

Object.values(searchSites).forEach((site) => {
  searchSlashCommand.addSubcommand((subcommand) =>
    subcommand
      .setName(site.name.toLowerCase())
      .setDescription(site.description)
      .addStringOption((option) =>
        option
          .setName('prompt')
          .setDescription('term to search')
          .setRequired(true),
      )
      .addUserOption((option) =>
        option.setName('user').setDescription('user to ping'),
      ),
  );
});
```
Factoring out the SlashCommandBuilder instantiation allows you to auto-apply any new search commands by only adding to the searchSites object, without needing to also remember to append `.addSubcommand(createCommand(searchSites[nameOfSite]))`.

Reply:
This was initially the idea, but it felt a bit unclear as the structure wasn't similar to the other commands.

Reply:
You could always crack out the ol' IIFE to keep the structure within the file in line with others, where the slash command is defined/modified directly within `.data` property.

Reply:
Could do, although with an array, you'd need to `.find` in the execute callback to get the right object as opposed to just accessing the object key.

Having just said that, the `name` field for each object is redundant given it's just the key without forced lowercase. What I used with `Objects.values` can be changed to keys or entries.

Reply:
Read the edit.

Reply:
Maps are lovely 👌